examples/tsc/BUILD.gn

```diff
@@ -0,0 +1,201 @@
+
+group("default") {
+  testonly = true
+  deps = [
+    ":tsc",
+  ]
+}
+
+main_extern = [
+  "$rust_build:ansi_term",
+  "$rust_build:atty",
+  "$rust_build:dirs",
+  "$rust_build:flatbuffers",
+  "$rust_build:futures",
+  "$rust_build:getopts",
+  "$rust_build:http",
+  "$rust_build:hyper",
+  "$rust_build:hyper_rustls",
+  "$rust_build:lazy_static",
+  "$rust_build:libc",
+  "$rust_build:log",
+  "$rust_build:rand",
+  "$rust_build:regex",
+  "$rust_build:remove_dir_all",
+  "$rust_build:ring",
+  "$rust_build:rustyline",
+  "$rust_build:serde_json",
+  "$rust_build:source_map_mappings",
+  "$rust_build:tempfile",
+  "$rust_build:tokio",
+  "$rust_build:tokio_executor",
+  "$rust_build:tokio_fs",
+  "$rust_build:tokio_io",
+  "$rust_build:tokio_process",
+  "$rust_build:tokio_threadpool",
+  "$rust_build:url",
+]
+if (is_win) {
+  main_extern += [ "$rust_build:winapi" ]
+}
+
+ts_sources = [
+  "main.ts",
+  "../../js/assets.ts",
+  "../../js/blob.ts",
+  "../../js/buffer.ts",
+  "../../js/chmod.ts",
+  "../../js/console_table.ts",
+  "../../js/compiler.ts",
+  "../../js/console.ts",
+  "../../js/copy_file.ts",
+  "../../js/custom_event.ts",
+  "../../js/deno.ts",
+  "../../js/dir.ts",
+  "../../js/dispatch.ts",
+  "../../js/dom_types.ts",
+  "../../js/errors.ts",
+  "../../js/event.ts",
+  "../../js/event_target.ts",
+  "../../js/fetch.ts",
+  "../../js/format_error.ts",
+  "../../js/dom_file.ts",
+  "../../js/file_info.ts",
+  "../../js/files.ts",
+  "../../js/flatbuffers.ts",
+  "../../js/form_data.ts",
+  "../../js/globals.ts",
+  "../../js/headers.ts",
+  "../../js/io.ts",
+  "../../js/libdeno.ts",
+  "../../js/lib.web_assembly.d.ts",
+  "../../js/location.ts",
+  "../../js/main.ts",
+  "../../js/make_temp_dir.ts",
+  "../../js/metrics.ts",
+  "../../js/mkdir.ts",
+  "../../js/mock_builtin.js",
+  "../../js/net.ts",
+  "../../js/os.ts",
+  "../../js/platform.ts",
+  "../../js/plugins.d.ts",
+  "../../js/process.ts",
+  "../../js/read_dir.ts",
+  "../../js/read_file.ts",
+  "../../js/read_link.ts",
+  "../../js/remove.ts",
+  "../../js/rename.ts",
+  "../../js/repl.ts",
+  "../../js/resources.ts",
+  "../../js/stat.ts",
+  "../../js/symlink.ts",
+  "../../js/text_encoding.ts",
+  "../../js/timers.ts",
+  "../../js/truncate.ts",
+  "../../js/types.ts",
+  "../../js/url.ts",
+  "../../js/url_search_params.ts",
+  "../../js/util.ts",
+  "../../js/workers.ts",
+  "../../js/write_file.ts",
+  "../../js/performance.ts",
+  "../../js/version.ts",
+  "../../tsconfig.json",
+
+  # Listing package.json and yarn.lock as sources ensures the bundle is rebuilt
+  # when npm packages are added/removed or their contents changes.
+  "../../package.json",
+  "../../third_party/yarn.lock",
+]
+
+# When Cargo is driving the build, GN/Ninja are used to produce these non-Rust
+# targets. Cargo handles all Rust source files and the final linking step.
+group("tsc_deps") {
+  deps = [
+    ":msg_rs",
+    ":snapshot_tsc",
+    "../../libdeno:libdeno_static_lib",
```

Comment:
Ideally we'd have `//:deno_deps` here

Reply:
The snapshots are unique, so I have reconfigured so that two common dependencies are in the group.